aws/s3/s3.go

```diff
@@ -21,9 +21,10 @@ import (
 )
 
 type S3 struct {
-	client     *s3.Client
-	uploader   *manager.Uploader
-	downloader *manager.Downloader
+	client       *s3.Client
+	uploader     *manager.Uploader
+	downloader   *manager.Downloader
+	concurrenter *ConcurrencyManager
```

Comment:
Maybe we define a `ConcurrentS3` like 
```
type ConcurrentS3 struct {
  S3
  concurrenter *ConcurrencyManager
}
```

Then all the functions goes into `ConcurrentS3` so the current S3 remains a simple wrapping of AWS SDK S3 package?